dcos_tunnel/cli.py

```diff
@@ -632,37 +654,49 @@ def _vpn(port, config_file, user, privileged, ssh_port, host,
 
         msg = "\nWaiting for VPN server in container '{}' to come up..."
         emitter.publish(msg.format(container_name))
+
         scom = ("until "
-                """[ "$(docker inspect --format='{{ .State.Running }}' """
+                """[ "$(%s inspect --format='{{ .State.Running }}' """
                 """%s 2>/dev/null)" = "true" ] 2>/dev/null; do sleep 0.5; """
-                """done""") % (container_name)
-        scom += (";"
-                 """docker exec {} "sh -c 'until [ -s {} ]; do sleep 0.5; """
-                 """done'" """).format(container_name, remote_keyfile)
-        scom += (";"
-                 """docker exec {} "sh -c 'until [ -s {} ]; do sleep 0.5; """
-                 """done'" """).format(container_name, remote_clientfile)
+                """done""") % (docker_cmd, container_name)
+        scom += (" && "
+                 """{} exec {} sh -c 'until [ -s {} ]; do sleep 0.5; """
+                 """done' """).format(docker_cmd, container_name,
+                                      remote_keyfile)
+        scom += (" && "
+                 """{} exec {} sh -c 'until [ -s {} ]; do sleep 0.5; """
+                 """done' """).format(docker_cmd, container_name,
+                                      remote_clientfile)
         server_setup = client.exec_command(scom, get_pty=True)
-        ssh_exec_wait(server_setup)
 
-        scom = ("""docker inspect --format='"""
+        if not vpn_server.is_alive():
```

Comment:
The `until` loop will hang because the server container never started in the first place. It's not the cost, but rather that regardless of the reason, if the container never comes up, or if the `docker` command can't run, the `until` loop will never terminate and we will never even encounter future checks.

If it gets past the `until` loop then the vpn server successfully started at some point in time. In that case, if it crashes, we will also exit here because the vpn client will fail to connect which is the intended behavior.